components/ingest-service/server/delete_nodes_scheduler.go

```diff
@@ -5,18 +5,25 @@ package server
 import (
 	"context"
 
-	"github.com/chef/automate/api/interservice/ingest"
+	log "github.com/sirupsen/logrus"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
-	log "github.com/sirupsen/logrus"
+	"github.com/chef/automate/api/interservice/ingest"
+	"github.com/chef/automate/lib/cereal"
 )
 
 // DeleteMarkedNodes - run the delete node task now
 func (server *JobSchedulerServer) DeleteMarkedNodes(ctx context.Context,
 	empty *ingest.DeleteMarkedNodesRequest) (*ingest.DeleteMarkedNodesResponse, error) {
-	jConfig := server.configManager.GetDeleteNodesSchedulerConfig()
-	deletedNodes, err := server.client.DeleteMarkedNodes(ctx, jConfig.Threshold)
+
+	var threshold string
+	err := server.jobManager.GetScheduledWorkflowParameters(ctx, "periodic_delete_nodes", "delete_nodes", &threshold)
```

Comment:
Strange pattern here. Why not just return the string as a value instead of giving a pointer to a string? By using a pointer we have to worry about it being `nil`. I don't see much in the way of a performance gain either because strings efficiently use pointers under the hood.

Reply:
You need a pointer here because GetShceduledWorkflowParameters basically unmarshal's the parameters into the pointer you give it and you can't unmarshal into a value.

Reply:
I'm thinking in another PR, I'm going to make a GetScheduledWorkflowByName() that returns a cereal.Schedule and then the caller can do the unmarshal themselves.  It is annoying as it is then a bit more code on the caller side, but would probably avoid this confusion here.

Reply:
Yes! What I was trying to describe (badly) in standup was more or less that kind of an approach, but in the end I wasn't sure if I could justify it. It adds more verbosity and more boilerplate but I think you might gain a bit more clarity.